modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationMetricsListener.java

```
@@ -247,10 +278,55 @@ public long receivedBytesCount() {
      * @return Map containing receiver nodes and respective counts.
      */
     public Map<UUID, Long> sentMessagesByNode() {
+        return collectMessagesCountByNodeId(SENT_MESSAGES_BY_NODE_ID_METRIC_NAME);
+    }
+
+    /** */
+    protected Map<String, Long> collectMessagesCountByType(String prefix) {
+        Map<String, Long> res = new HashMap<>();
+
+        prefix = MetricUtils.metricName(COMMUNICATION_METRICS_GROUP_NAME, prefix);
+
+        for (Metric metric : mreg) {
+            if (metric.name().startsWith(prefix)) {
+                try {
+                    short directType = Short.parseShort(metric.name().substring(prefix.length()));
+
+                    Map<Short, String> msgTypMap0 = msgTypMap;
+
+                    if (msgTypMap0 != null) {
+                        String typeName = msgTypMap0.get(directType);
+
+                        if (typeName != null)
+                            res.put(typeName, ((LongMetric)metric).value());
+                    }
+                }
+                catch (NumberFormatException ignore) {
+                }
+            }
+        }
+
+        return res;
+    }
+
+    /** */
+    protected Map<UUID, Long> collectMessagesCountByNodeId(String metricName) {
         Map<UUID, Long> res = new HashMap<>();
 
-        for (ThreadMetrics metrics : allMetrics)
-            addMetrics(res, metrics.sentMsgsCntByNode);
+        String mregPrefix = COMMUNICATION_METRICS_GROUP_NAME + MetricUtils.SEPARATOR;
+        for (MetricRegistry mreg : mmgr) {
+            if (mreg.name().startsWith(mregPrefix)) {
+                String nodeIdStr = mreg.name().substring(mregPrefix.length());
+
+                try {
+                    UUID nodeId = UUID.fromString(nodeIdStr);
+
+                    res.put(nodeId, mreg.<LongMetric>findMetric(metricName).value());
+                }
+                catch (IllegalArgumentException ignore) {
```

We shouldn't swallow exception here. Let's remove empty catch block.